conans/test/unittests/model/build_info/components_test.py

```diff
@@ -0,0 +1,178 @@
+# coding=utf-8
+import os
+import unittest
+import warnings
+
+from conans.model.build_info import CppInfo, DepsCppInfo, DepCppInfo
+from conans.test.utils.test_files import temp_folder
+
+
+class CppInfoComponentsTest(unittest.TestCase):
+
+    def test_components_set(self):
+        cpp_info = CppInfo("root_folder")
+        cpp_info.components["liba"].name = "LIBA"
+        cpp_info.components["libb"].includedirs.append("includewhat")
+        cpp_info.components["libc"].libs.append("thelibc")
+        self.assertListEqual(list(cpp_info.components.keys()), ["liba", "libb", "libc"])
+        self.assertEqual(cpp_info.components["liba"].name, "LIBA")
+        self.assertListEqual(cpp_info.components["libb"].includedirs, ["include", "includewhat"])
+        self.assertListEqual(cpp_info.components["libc"].libs, ["thelibc"])
+
+    def test_no_components_inside_components(self):
+        cpp_info = CppInfo("root_folder")
+        cpp_info.components["liba"].name = "LIBA"
+        with self.assertRaises(AttributeError):
+            cpp_info.components["libb"].components["libb"].name = "LIBB"
```

Comment:
``cpp_info.components["libb"].components`` should be enough to raise the exception?